File: app/src/test/java/com/example/notphilphil/bob/controllers/TestLoggedUser.java
package com.example.notphilphil.bob.controllers;

import com.example.notphilphil.bob.models.User;

import org.junit.Test;

import static junit.framework.Assert.assertEquals;
import static junit.framework.Assert.assertFalse;
import static junit.framework.Assert.assertTrue;

public class TestLoggedUser {

    @Test
    public void testLogout() {
        LoggedUser.setTesting(true);
        String id = "id";
        String name = "name";
        LoggedUser.newInstance(new User(id, name));

        // Check if logged in
        assertEquals(id, LoggedUser.getId());
        assertEquals(name, LoggedUser.getName());
        assertEquals(LoggedUser.PermissionsEnum.USER, LoggedUser.getPermissions());
        assertTrue(LoggedUser.isLoggedIn());

        LoggedUser.logOut();

        // Check if logged out
        assertEquals("", LoggedUser.getId());
        assertEquals("", LoggedUser.getName());
        assertEquals(LoggedUser.PermissionsEnum.NONE, LoggedUser.getPermissions());
        assertFalse(LoggedUser.isLoggedIn());
    }
}

File: app/src/test/java/com/example/notphilphil/bob/controllers/TestLoggedUser.kt
package com.example.notphilphil.bob.controllers

import com.example.notphilphil.bob.models.User
import com.google.firebase.FirebaseApp

import org.junit.Test

import junit.framework.Assert.assertEquals
import junit.framework.Assert.assertFalse
import junit.framework.Assert.assertTrue

class TestLoggedUser {

    @Test
    fun testLogout() {
        //LoggedUser.setTesting(true);
        val id = "id"
        val name = "name"
        LoggedUser.newInstance(User(id, name))

        // Check if logged in
        assertEquals(id, LoggedUser.id)
        assertEquals(name, LoggedUser.name)
        assertEquals(LoggedUser.PermissionsEnum.USER, LoggedUser.permissions)
        assertTrue(LoggedUser.isLoggedIn)

        LoggedUser.logOut()

        // Check if logged out
        assertEquals("", LoggedUser.id)
        assertEquals("", LoggedUser.name)
        assertEquals(LoggedUser.PermissionsEnum.NONE, LoggedUser.permissions)
        assertFalse(LoggedUser.isLoggedIn)
    }
}
